### PR TITLE
[Lang] Support simple matrix slicing

### PR DESCRIPTION
**Pull request by mzmzm**
Related issue = #4257
Simple implementation of matrix slicing, that only supports the positive constant integer type of slice's elements.

### CODE REVIEW

File: python/taichi/lang/impl.py [hunk 1]
```diff
@@ -126,23 +126,34 @@ def wrap_scalar(x):
 @taichi_scope
 def subscript(value, *_indices, skip_reordered=False):
     if isinstance(value, np.ndarray):
-        return value.__getitem__(*_indices)
+        return value.__getitem__(_indices)
```
Comment (strongoier) [inline python/taichi/lang/impl.py:129]: Why this change?

File: tests/python/test_simple_matrix_slice.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        c = ti.Vector([0, 1, 2, 3, 4, 5, 6])
+        return c[:5:2]
+
+    @ti.kernel
+    def foo2() -> ti.types.matrix(2, 2, dtype=ti.i32):
+        a = ti.Matrix([[1, 2, 3], [4, 5, 6]])
+        return a[:, :b:2]
+
+    v1 = foo1()
+    assert v1[0] == 0
```
Comment (strongoier) [inline tests/python/test_simple_matrix_slice.py:20]: Shall we verify the full matrix/vector instead of a single value?

**Review by strongoier — COMMENTED** [review]
Nice work!

File: python/taichi/lang/impl.py (hunk 1 above)
Comment (mzmzm) [reply python/taichi/lang/impl.py:129]: `ndarray.__getitem__` only supports an index or a tuple of indices, the previous one has a bug then deal with multiple indexes

File: python/taichi/lang/impl.py [hunk 3]
```diff
@@ -126,23 +126,34 @@ def wrap_scalar(x):
...
     _indices = tuple(flattened_indices)
     if isinstance(_indices,
                   tuple) and len(_indices) == 1 and _indices[0] is None:
         _indices = ()
+
+    if has_slice:
+        if not isinstance(value, Matrix):
+            raise SyntaxError(
+                f"The type {type(value)} do not support index of slice type")
+        return value._subscript(*_indices)
```
[the remark this answers is not in the document]
Comment (mzmzm) [reply python/taichi/lang/impl.py]: fix

File: tests/python/test_simple_matrix_slice.py (hunk 2 above)
Comment (mzmzm) [reply tests/python/test_simple_matrix_slice.py]: fix

**Review by strongoier — APPROVED** [review]
LGTM!
